dace/codegen/instrumentation/data/data_dump.py

```diff
@@ -1,204 +1,205 @@
+
+    def __init__(self):
+        super().__init__()
+        self.gpu_runtime_init = False
+        from dace.codegen.targets.framecode import DaCeCodeGenerator  # Avoid import loop
+        self.codegen: DaCeCodeGenerator = None
+
+    def _generate_report_setter(self, sdfg: SDFG) -> str:
+        return f'''
+        DACE_EXPORTED void __dace_set_instrumented_data_report({sdfg.name}_t *__state, const char *dirpath) {{
+            __state->serializer->set_folder(dirpath);
+        }}
+        '''
+
+    def on_sdfg_begin(self, sdfg: SDFG, local_stream: CodeIOStream, global_stream: CodeIOStream,
+                      codegen: 'DaCeCodeGenerator'):
+        # Initialize serializer versioning object
+        if sdfg.parent is None:
+            self.codegen = codegen
+            codegen.statestruct.append('dace::DataSerializer *serializer;')
+            sdfg.append_init_code(f'__state->serializer = new dace::DataSerializer("");\n')
+
+            # Add method that controls serializer input
+            global_stream.write(self._generate_report_setter(sdfg))
+
+    def on_sdfg_end(self, sdfg: SDFG, local_stream: CodeIOStream, global_stream: CodeIOStream):
+        # Teardown serializer versioning object
+        if sdfg.parent is None:
+            sdfg.append_exit_code('delete __state->serializer;\n')
+
+    def on_node_begin(self, sdfg: SDFG, state: SDFGState, node: nodes.AccessNode, outer_stream: CodeIOStream,
+                      inner_stream: CodeIOStream, global_stream: CodeIOStream):
+        from dace.codegen.dispatcher import DefinedType  # Avoid import loop
+
+        if is_devicelevel_gpu(sdfg, state, node) or is_devicelevel_fpga(sdfg, state, node):
+            # Only run on host code
+            return
+
+        desc = node.desc(sdfg)
+
+        # Obtain a pointer for arrays and scalars
+        ptrname = cpp.ptr(node.data, desc, sdfg, self.codegen)
+        defined_type, _ = self.codegen.dispatcher.defined_vars.get(ptrname)
+        if defined_type == DefinedType.Scalar:
+            ptrname = '&' + ptrname
+
+        # Create UUID
+        state_id = sdfg.node_id(state)
+        node_id = state.node_id(node)
+        uuid = f'{sdfg.sdfg_id}_{state_id}_{node_id}'
+
+        # Get optional pre/postamble for instrumenting device data
+        preamble, postamble = '', ''
+        if desc.storage == dtypes.StorageType.GPU_Global:
+            self._setup_gpu_runtime(sdfg, global_stream)
+            preamble, postamble, ptrname = self._generate_copy_to_device(node, desc, ptrname)
+
+        # Write code
+        inner_stream.write(preamble, sdfg, state_id, node_id)
+        inner_stream.write(
+            f'__state->serializer->restore({ptrname}, {cpp.sym2cpp(desc.total_size - desc.start_offset)}, '
+            f'"{node.data}", "{uuid}");\n', sdfg, state_id, node_id)
+        inner_stream.write(postamble, sdfg, state_id, node_id)
```

Comment:
Did anything change in this file apart from adding `from dace.codegen import common` in line 8?

Reply:
Yes, it gets the backend with the new function